app/controllers/api/v0/persons_controller.rb

```diff
@@ -17,7 +17,7 @@ def index
   def show
     person = Person.current.includes(:user, :ranksSingle, :ranksAverage).find_by_wca_id!(params[:wca_id])
     private_attributes = []
-    if current_user && current_user.can_admin_results?
+    if current_user && (current_user.can_admin_results? || current_user&.person == person)
```

Comment:
`current_user.person` is by definition identified by the User's `wca_id`. You also have an ID in `params[:wca_id]`. So doing `current_user.wca_id == params[:wca_id]` is a lot cheaper and efficient to execute